### PR TITLE
Bento Carousel: `outsetArrows` prop

### PR DESCRIPTION
**Pull request by caroqliu**
This PR adds the `outsetArrows` prop to `amp-base-carousel` with unit tests. 

`outsetArrows` = `Boolean`
- Whether to display next and prev arrows beside the carousel (`true`) or on top of it (`false`). Defaults to `false`. 
- When `true`, there are the following additional side effects: 
  - The width of the slides is `100px` less than the width of the full carousel to account for the arrows (`50px` each). 
  - `controls` attribute is ignored and defaults to value `"always"`
- `true`:
image

- `false`:
image

Note this prop was originally exclusive to `amp-stream-gallery` but is being moved to `amp-base-carousel` instead. Partial for #28284

### CODE REVIEW

File: extensions/amp-base-carousel/1.0/base-carousel.jss.js [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import {createUseStyles} from 'react-jss';
 
 const scrollContainer = {
-  position: 'absolute',
+  position: 'relative',
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/1.0/base-carousel.jss.js:20]: If it's relative now, should we remove `top/left/right/bottom: 0`?

File: extensions/amp-base-carousel/1.0/base-carousel.type.js [hunk 2]
```diff
@@ -59,7 +59,8 @@ BaseCarouselDef.SlideProps;
  *   advance: (function(number):undefined|undefined),
  *   customArrow: (PreactDef.VNode|undefined),
  *   by: number,
- *   disabled: (boolean|undefined)
+ *   disabled: (boolean|undefined),
+ *   outsetArrows: (boolean|undefined)
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/1.0/base-carousel.type.js:63]: Nit: trailing commas are allowed here as well.

**Review by dvoytenko — APPROVED** [review]
LGTM with a couple of questions.

File: extensions/amp-base-carousel/1.0/base-carousel.jss.js (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/base-carousel.jss.js:20]: Good catch, done.

File: extensions/amp-base-carousel/1.0/base-carousel.type.js (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/base-carousel.type.js]: Done.

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**